velox/vector/fuzzer/VectorFuzzer.cpp

```diff
@@ -607,4 +655,80 @@ RowTypePtr VectorFuzzer::randRowType(int maxDepth) {
   return ROW(std::move(names), std::move(fields));
 }
 
+VectorPtr VectorFuzzer::wrapInLazyVector(VectorPtr baseVector) {
+  return std::make_shared<LazyVector>(
+      baseVector->pool(),
+      baseVector->type(),
+      baseVector->size(),
+      std::make_unique<VectorLoaderWrap>(baseVector));
+}
+
+RowVectorPtr VectorFuzzer::fuzzRowChildrenToLazy(RowVectorPtr rowVector) {
+  std::vector<VectorPtr> children;
+  VELOX_CHECK_NULL(rowVector->nulls());
+  for (auto child : rowVector->children()) {
+    VELOX_CHECK_NOT_NULL(child);
+    VELOX_CHECK(!child->isLazy());
+    // TODO: If child has dictionary wrappings, add ability to insert lazy wrap
+    // between those layers.
+    children.push_back(coinToss(0.5) ? wrapInLazyVector(child) : child);
+  }
+  return std::make_shared<RowVector>(
+      pool_,
+      rowVector->type(),
+      nullptr,
+      rowVector->size(),
+      std::move(children));
+}
+
+VectorPtr VectorLoaderWrap::makeEncodingPreservedCopy(SelectivityVector& rows) {
+  VectorPtr result;
+  DecodedVector decoded;
+  decoded.decode(*vector_, rows, false);
+
+  if (decoded.isConstantMapping() || decoded.isIdentityMapping()) {
+    BaseVector::ensureWritable(rows, vector_->type(), vector_->pool(), result);
+    result->copy(vector_.get(), rows, nullptr);
+    return result;
+  }
+
+  SelectivityVector baseRows;
+  auto baseVector = decoded.base();
+
+  baseRows.resize(baseVector->size(), false);
```

Comment:
I remember seeing a more efficient API to copy these nullability bitmaps without having an explicit loop.